scalding-core/src/main/scala/com/twitter/scalding/estimation/memory/SmoothedHistoryMemoryEstimator.scala

```diff
@@ -0,0 +1,117 @@
+package com.twitter.scalding.estimation.memory
+
+import com.twitter.scalding.estimation.{ FlowStepHistory, FlowStrategyInfo, HistoryEstimator, Task }
+import org.apache.hadoop.mapred.JobConf
+import org.slf4j.LoggerFactory
+
+// Tuple(MapMemory in MB for java process and container, ReduceMemory in MB for java process and container),
+// or None to keep the default.
+case class MemoryEstimate(mapMemoryInMB: Option[(Long, Long)], reduceMemoryInMB: Option[(Long, Long)])
+
+object SmoothedHistoryMemoryEstimator {
+  val CommittedHeapBytes = "COMMITTED_HEAP_BYTES"
+  val CpuMs = "CPU_MILLISECONDS"
+  val PhysicalMemoryBytes = "PHYSICAL_MEMORY_BYTES"
+  val GCTimeMs = "GC_TIME_MILLIS"
```

Comment:
there should be some configurable max like 5% for GCTImeMs/CpuMs to correct under-provisioned Xmx.